### PR TITLE
Add compiler warning for uninitialized variable and fix

### PR DESCRIPTION
**Pull request by wenkaidu**
## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._
SWDEV-455705

**What were the changes?**  
_One sentence describing the work done._

**Why were the changes made?**  
Fix an UB that could lead to miscompilation

**How was the outcome achieved?**  
Add compiler warning for uninitialized variable and fix

**Additional Documentation:**  
SWDEV-455705

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

**Review by nileshnegi — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -598,6 +598,7 @@ endif()
 
 ## Set RCCL compile options
 target_compile_options(rccl PRIVATE -parallel-jobs=12)
+target_compile_options(rccl PRIVATE -Wuninitialized)
```
Comment (shiltian) [inline CMakeLists.txt]: It might be better to add an extra one `-Wsometimes-uninitialized`.

Comment (wenkaidu) [reply CMakeLists.txt]: done

**Review by shiltian — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -598,6 +598,7 @@ endif()
 
 ## Set RCCL compile options
 target_compile_options(rccl PRIVATE -parallel-jobs=12)
+target_compile_options(rccl PRIVATE -Wuninitialized -Wsometimes-uninitialized)
```
Comment (shiltian) [inline CMakeLists.txt]: Furthermore, if you want to turn this kind of warnings into errors such that they can be easily identified, you can use `-Werror=uninitialized -Werror=sometimes-uninitialized`. https://godbolt.org/z/xGejP6en5

Comment (wenkaidu) [reply CMakeLists.txt]: done

**Pull request merged**